src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java

```diff
@@ -393,14 +412,53 @@ private static <T extends Exprent> void findExprents(Statement start, Class<? ex
    *        // code for case "foo"
    *   }
    */
-  private static boolean isSwitchOnString(SwitchStatement first, SwitchStatement second) {
+  private static boolean isSwitchOnString(SwitchStatement first) {
+    SwitchStatement second = null;
+    List<StatEdge> edges = first.getSuccessorEdges(StatEdge.TYPE_REGULAR);
+    if (edges.size() == 1 && edges.get(0).getDestination().type == Statement.TYPE_SWITCH) {
+      second = (SwitchStatement)edges.get(0).getDestination();
+    }
+    AssignmentExprent nullAssign = null;
+
+    // if we're the only thing in an if statement,
+    if (first.getParent() instanceof IfStatement) {
+      if (first.getSuccessorEdges(StatEdge.TYPE_REGULAR).size() == 0) {
+        IfStatement parent = (IfStatement) first.getParent();
+        Exprent ifCond = parent.getHeadexprent().getCondition();
+        // and it's a null check,
+        if (ifCond instanceof FunctionExprent) {
+          FunctionExprent func = (FunctionExprent)ifCond;
+          if (func.getFuncType() == FunctionExprent.FUNCTION_NE && func.getLstOperands().size() == 2) {
+            Exprent right = func.getLstOperands().get(1);
+            if (right.type == Exprent.EXPRENT_CONST && right.getExprType() == VarType.VARTYPE_NULL) {
+              // and the `else` only assigns a variable,
+              Statement elseStat = parent.getElsestat();
```

Comment:
getElsestat() will be null unless IfStatement#getIftype() == IfStatement.IFTYPE_IFELSE

Reply:
The `instanceof` immediately after checks for null - should I add an explicit if-type check anyways?

Reply:
oh right- yea I think you should

Reply:
done